graph/hypergraph.py

```diff
@@ -41,6 +43,21 @@ def split_edge(self, edge: Edge) -> Node:
 
         return new_node
 
+    def create_center_node(self, nodes: list[Node]) -> Node:
+        new_x = np.average([node.x for node in nodes]).astype(float)
+        new_y = np.average([node.y for node in nodes]).astype(float)
+        new_node = Node(x=new_x, y=new_y, is_hanging=False)
+        self.nodes.append(new_node)
+
+        return new_node
+
+    def create_edges(self, central_node: Node, nodes: list[Node], is_border: bool = False) -> None:
```

Comment:
rename, so its easier to understand what it does based only on name (or add description)